docs/decisions/0006-api-specification-for-enterprise-mfes.rst

```diff
@@ -0,0 +1,255 @@
+0006 API Specification for Enterprise Micro-frontends (MFEs)
+************************************************************
+
+Status
+======
+
+Accepted (March 2023)
+
+Context
+=======
+
+This document intends to outline which API endpoints exist or will exist in support of the
+Enterprise MFEs (e.g., frontend-app-learner-portal-enterprise) as well as other consumers,
+or service-to-service communication.
+
+API Endpoints
+=============
+
+GET Retrieve single, redeemable access policy for a course
+------------------------------------------------------
+
+**/api/v1/enterprise-customer/<enterprise_customer_uuid>/policy/can_redeem/**
+
+This API endpoint will be called by the enterprise learner portal to understand whether
+the learner is already enrolled in the course (i.e., a prior redemption has been successfully
+fulfilled) and/or which subsidy access policy should be used to redeem the course when a learner
+clicks the "Enroll" button.
+
+At a high level, this API endpoint works by iterating over all subsidy access policies associated with
+the enterprise customer for the specified learner to understand which subsidy access policies are indeed
+redeemable. It does this by calling the ``can_redeem`` API endpoint in ``enterprise-subsidy`` for each active
+subsidy access policy. In the event multiple subsidy access policies are found, this API endpoint chooses
+the preferred subsidy basked on our business rules.
+
+This API endpoint differs from the one spec'd in the 0003 Initial API Specification in that
+it makes a decision of which single subsidy access policy should be redeemed for a given course in the event
+a learner has multiple redeemable subsidy access policies. The API endpoint that's already spec'd returns a
+list of redeemable policies, which would mean the client  (e.g., enterprise learner portal) still requires business
```

Comment:
> The API endpoint that's already spec'd returns a list of redeemable policies, which would mean the client 

I'm wondering if instead of a whole new endpoint perhaps we should just pass a query parameter can_redeem to only return the valid policy that should be used? `show_only_valid_policy` or something of the like? 

My thought process here being that these endpoints basically do the same thing but only differ in their output... I dunno I just don't want the routing to get weird.